Add duplicate keys lint (#250)
Kampfkarren/selene: docs/src/lints/duplicate_keys.md

# duplicate_keys
## What it does
Checks for duplicate keys being defined inside of tables.

## Why this is bad

> Tables with a key defined more than once can lead to unexpected behaviour in your program.

Reviewer: Let's be more specific.

```suggestion
Tables with a key defined more than once will only use one of the values.
```
